src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc

```diff
@@ -110,14 +111,12 @@ void AddSubMenuRequest::Run() {
                                  [strings::value].asString(),
                       app,
                       application_manager_);
-
     msg_params[strings::sub_menu_icon] =
         (*message_)[strings::msg_params][strings::sub_menu_icon];
-    SendHMIRequest(hmi_apis::FunctionID::UI_AddSubMenu, &msg_params, true);
   } else {
-    LOG4CXX_ERROR(logger_, "Sub-menu icon is not valid.");
-    SendResponse(false, mobile_apis::Result::INVALID_DATA);
+     msg_params.erase(strings::sub_menu_icon);
```

Comment:
@pold500 If the condition in the "if" statement in line 108 includes is_dynamic_image == TRUE, you will delete the icon image parameter if it is STATIC, but I think you don't have to do this. I recommend you after you have checked that the image parameter exists, check if it is dynamic (in another if clause) and if it is - add full icon path. If it is not - do not change mgs_params (do nothing). If you need some help or my comment is not fully understandable, please call me.